geonode/proxy/views.py

```diff
@@ -104,52 +101,24 @@ def proxy(
     if url.fragment != "":
         locator += f"#{url.fragment}"
 
-    # White-Black Listing Hosts
-    site_url = urlsplit(settings.SITEURL)
-    if sec_chk_hosts and not settings.DEBUG:
-        # Attach current SITEURL
-        if site_url.hostname not in PROXY_ALLOWED_HOSTS:
-            PROXY_ALLOWED_HOSTS += (site_url.hostname,)
-
-        # Attach current hostname
-        hostname = (ogc_server_settings.hostname,) if ogc_server_settings else ()
-        if hostname not in PROXY_ALLOWED_HOSTS:
-            PROXY_ALLOWED_HOSTS += hostname
-
-        # Check OWS regexp
-        if url.query and ows_regexp.match(url.query):
-            ows_tokens = ows_regexp.match(url.query).groups()
-            if (
-                len(ows_tokens) == 4
-                and "version" == ows_tokens[0]
-                and StrictVersion(ows_tokens[1]) >= StrictVersion("1.0.0")
-                and StrictVersion(ows_tokens[1]) <= StrictVersion("3.0.0")
-                and ows_tokens[2].lower() in ("getcapabilities")
-                and ows_tokens[3].upper() in ("OWS", "WCS", "WFS", "WMS", "WPS", "CSW")
+    if sec_chk_hosts:
+        if url.hostname not in proxy_urls_registry.get_proxy_allowed_hosts():
+            if not any(needle.lower() in url.query.lower() for needle in PROXY_ALLOWED_PARAMS_NEEDLES) and not any(
+                needle.lower() in url.path.lower() for needle in PROXY_ALLOWED_PATH_NEEDLES
```

Comment:
@giohappy Does it make sense to allow hosts to be proxied, just because some paths or query parameters are allowed? As an administrator, I would like to configure allowed hosts explicitly. A bad URL (maybe from a user input) which contains an arbitrary needle (path or query does not matter here) would be allowed by the proxy.

I see `PROXY_ALLOWED_PARAMS_NEEDLES` and `PROXY_ALLOWED_PATH_NEEDLES` are configurable, but these match for all hosts which are actually /not/ allowed.

IMO, servers which expose a Web API (e.g. OGC WMS) which is not CORS enabled (I guess, this is what this is all about) is not configured correctly. Not sure, if the operator wants all traffic to be routed through the GeoNode proxy. That would seem to me more as a quick and dirty workaround rather than a good productive configuration. 

What do you think? Did I miss something?

Reply:
hi @ridoo I understand your concerns, and I agree that this solution is far from ideal. 

This has been implemented to meet the logic that MS implements now. The client allows adding external layers to a map, and it implements the reverse logic compared to GeoNode: **if the URLs are not listed inside a `useCORS` array (that GeoNode sets empty) they are automatically proxied**.  Basically [it does](https://github.com/geosolutions-it/MapStore2/blob/master/java/web/src/main/resources/proxy.properties#L26-L30) what we're doing here.

If we do not support this, the catalog functionality that will be available since GeoNode 4.3.0 (thanks to a more extended integration with MS) won't work.
This is a functionality that many users wanted from MS, and this is why we are making this (temporary) move.
The final solution will be to make MS support the reverse logic, which is the one implemented by GeoNode until now: **it should go straight to the origin server, unless stated differently (`PROXY_ALLOWED_HOSTS`).**.

BTW, the two settings (which [will be set by the client](https://github.com/GeoNode/geonode-mapstore-client/commit/d6b63464b3550508459091aca94e3b2c34ec106f)) let administrators change this behavior.

Reply:
@ridoo The only alternative for now is to make the two settings an opt-in. However also the MS catalog plugin should be an opt-in in that case, otherwise, we will have a tool that will never work.

Reply:
Hey @giohappy, so this is more a (temporary) compatibility thing. Thanks for the background.

Reply:
>Hey @giohappy, so this is more a (temporary) compatibility thing. Thanks for the background.

correct @ridoo.